### PR TITLE
KAFKA-17816: Add testRuntimeOnly runtimeTestLibs to test-common-api module

### PR DESCRIPTION
**Pull request by mingyen066**
# Changes
* Add `testRuntimeOnly runtimeTestLibs` to `test-common-api` module
* Use `implementation libs.junitJupiterApi` instead of `implementation libs.junitJupiter` in module `test-common-api`

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by chia7712 — APPROVED** [review]

File: build.gradle [hunk 1]
```diff
@@ -1612,9 +1612,12 @@ project(':test-common:test-common-api') {
     implementation project(':server')
     implementation project(':server-common')
     implementation project(':test-common')
-    implementation libs.junitJupiter
+    implementation libs.junitJupiterApi
+
     testImplementation libs.mockitoCore
+
     testRuntimeOnly libs.junitPlatformLanucher
```
Comment (chia7712) [inline build.gradle]: we don't need to declare `junitPlatformLanucher` again, right?

File: build.gradle [hunk 2]
```diff
@@ -1612,9 +1612,12 @@ project(':test-common:test-common-api') {
     implementation project(':server')
     implementation project(':server-common')
     implementation project(':test-common')
-    implementation libs.junitJupiter
```
Comment (chia7712) [inline build.gradle:1615]: Please add `libs.junitJupiter` to `testImplementation`

File: build.gradle (hunk 1 above)
Comment (mingyen066) [reply build.gradle]: That's right.

**Review by m1a2st — APPROVED** [review]
@mingyen066 Thanks for the fix, LGTM

**Review by chia7712 — APPROVED** [review]

**Pull request merged**